js/services/http.js

```diff
@@ -100,6 +100,18 @@ define(function(require, exports) {
     }
   }
 
+  class FhirApi extends Api {
```

Comment:
I don't think FhirApi should be a nested class in http.js.  The http.js is a module which facilitates http communication.  I think the FHIR class handles communicating to a FHIR service.   We have a separate folder of `services` which you can implement your class there, and then you can import the class in the javascript compoennt you want to use (via the define())

Reply:
Apologies, my understanding about the http.js module was that it just wrapped some lower-level HTTP calls to handle passing specific headers, but not something that made some kind of 'client api' to a specific type of service (like the FHIR one or the PlainText one)...I'm not sure what the design decision was about introducing the PlainText api into http.js but it seems that whoever put that forward thought that you would put these type of API clients directly into the http.js module...so, I can't argue with that, so my only comment then is how the JSON_RESPONSE_TYPE  was removed....although not sure where that is being used (but possibly it's a constant that is refernced elsewhere in the app).